go/vt/vtctl/grpcvtctldserver/server_test.go

```diff
@@ -259,6 +267,185 @@ func TestGetTablet(t *testing.T) {
+	vtctld := NewVtctldServer(ts)
+
+	validAlias := &topodatapb.TabletAlias{
+		Cell: "zone1",
+		Uid:  100,
+	}
+	testutil.AddTablet(ctx, t, ts, &topodatapb.Tablet{
+		Alias: validAlias,
+	})
+	otherAlias := &topodatapb.TabletAlias{
+		Cell: "zone1",
+		Uid:  101,
+	}
+	testutil.AddTablet(ctx, t, ts, &topodatapb.Tablet{
+		Alias: otherAlias,
+	})
+
+	// we need to run this on each test case or they will pollute each other
+	setupSchema := func() {
+		testutil.TabletManagerClient.Schemas[topoproto.TabletAliasString(validAlias)] = &tabletmanagerdatapb.SchemaDefinition{
+			DatabaseSchema: "CREATE DATABASE vt_testkeyspace",
+			TableDefinitions: []*tabletmanagerdatapb.TableDefinition{
+				{
+					Name: "t1",
+					Schema: `CREATE TABLE t1 (
+	id int(11) not null,
+	PRIMARY KEY (id)
+);`,
+					Type:       "BASE",
+					Columns:    []string{"id"},
+					DataLength: 100,
+					RowCount:   50,
+					Fields: []*querypb.Field{
+						{
+							Name: "id",
+							Type: querypb.Type_INT32,
+						},
+					},
+				},
+			},
+		}
+	}
+
+	tests := []*struct {
+		name      string
+		req       *vtctldatapb.GetSchemaRequest
+		expected  *vtctldatapb.GetSchemaResponse
+		shouldErr bool
+	}{
+		{
+			name: "normal path",
+			req: &vtctldatapb.GetSchemaRequest{
+				TabletAlias: validAlias,
+			},
+			expected: &vtctldatapb.GetSchemaResponse{
+				Schema: &tabletmanagerdatapb.SchemaDefinition{
+					DatabaseSchema: "CREATE DATABASE vt_testkeyspace",
+					TableDefinitions: []*tabletmanagerdatapb.TableDefinition{
+						{
+							Name: "t1",
+							Schema: `CREATE TABLE t1 (
+	id int(11) not null,
+	PRIMARY KEY (id)
+);`,
+							Type:       "BASE",
+							Columns:    []string{"id"},
+							DataLength: 100,
+							RowCount:   50,
+							Fields: []*querypb.Field{
+								{
+									Name: "id",
+									Type: querypb.Type_INT32,
+								},
+							},
+						},
+					},
+				},
+			},
+			shouldErr: false,
+		},
+		{
+			name: "table names only",
+			req: &vtctldatapb.GetSchemaRequest{
+				TabletAlias:    validAlias,
+				TableNamesOnly: true,
+			},
+			expected: &vtctldatapb.GetSchemaResponse{
+				Schema: &tabletmanagerdatapb.SchemaDefinition{
+					DatabaseSchema: "CREATE DATABASE vt_testkeyspace",
+					TableDefinitions: []*tabletmanagerdatapb.TableDefinition{
+						{
+							Name: "t1",
+						},
+					},
+				},
+			},
+			shouldErr: false,
+		},
+		{
+			name: "table sizes only",
+			req: &vtctldatapb.GetSchemaRequest{
+				TabletAlias:    validAlias,
+				TableSizesOnly: true,
+			},
+			expected: &vtctldatapb.GetSchemaResponse{
+				Schema: &tabletmanagerdatapb.SchemaDefinition{
+					DatabaseSchema: "CREATE DATABASE vt_testkeyspace",
+					TableDefinitions: []*tabletmanagerdatapb.TableDefinition{
+						{
+							Name:       "t1",
+							Type:       "BASE",
+							DataLength: 100,
+							RowCount:   50,
+						},
+					},
+				},
+			},
+			shouldErr: false,
+		},
+		{
+			name: "table names take precedence over table sizes",
```

Comment:
Small question to double check my read of this. 😊 

The `command/schema.go` implementation [throws an error](https://github.com/vitessio/vitess/pull/7346/files#diff-1c5c22fad2ed84bcb4eb95cfba78937f5d4ba268157b35abe9671896f300b842R49) if both are specified. For what it's worth, explicitly erroring when both flags are set seems preferable to (since it's less surprising than) implicit precedence... and I have a feeling that's why you wrote `command/schema.go` to error, right? 

And I have a feeling that we can't return an error here, and instead have precedence behaviour, because legacy reasons? 

To be clear, I think the discrepancy between the two is fine, and what you have is great. I just want to make sure I follow the difference. :)

Reply:
I think this isn't about maintaining "legacy" compatibility, but more that the options are fundamentally incompatible with each other. If I could have made the parser itself enforce the mutual exclusion, that would have been my highest preference.

I think there's an argument to be made that the server implementation itself should error in that case (instead of just relying on the CLI to catch it), but I think I would disagree on "be permissive in what you accept" kinds of reasons.

Reply:
That makes sense! I think the behaviour as you've written it is really well documented, between the protobuf comments + tests, so "surprising" was probably the wrong word. 😆 Thank you for explaining your thoughts; they are always good. :)